Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/conceptcollection/ConceptCollectionManagerTest.java

```diff
@@ -0,0 +1,218 @@
+package edu.asu.spring.quadriga.service.impl.conceptcollection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.codehaus.jettison.json.JSONException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.web.client.RestTemplate;
+
+import edu.asu.spring.quadriga.dao.conceptcollection.IConceptCollectionDAO;
+import edu.asu.spring.quadriga.dao.workspace.IListWsDAO;
+import edu.asu.spring.quadriga.domain.conceptcollection.IConcept;
+import edu.asu.spring.quadriga.domain.conceptcollection.IConceptCollection;
+import edu.asu.spring.quadriga.domain.conceptcollection.IConceptCollectionCollaborator;
+import edu.asu.spring.quadriga.domain.factory.conceptcollection.IConceptFactory;
+import edu.asu.spring.quadriga.domain.impl.ConceptpowerReply;
+import edu.asu.spring.quadriga.domain.impl.conceptcollection.Concept;
+import edu.asu.spring.quadriga.domain.impl.conceptcollection.ConceptCollection;
+import edu.asu.spring.quadriga.domain.impl.conceptcollection.ConceptCollectionCollaborator;
+import edu.asu.spring.quadriga.domain.impl.workbench.Project;
+import edu.asu.spring.quadriga.domain.impl.workspace.WorkSpace;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.domain.workspace.IWorkSpace;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.conceptcollection.mapper.IConceptCollectionDeepMapper;
+import edu.asu.spring.quadriga.service.workbench.mapper.IProjectShallowMapper;
+import edu.asu.spring.quadriga.service.workspace.IListWSManager;
+
+public class ConceptCollectionManagerTest {
+
+	@Mock
+	private RestTemplate mockedRestTemplate = Mockito.mock(RestTemplate.class);
+
+	@Mock
+	private IConceptCollectionDeepMapper mockedConceptCollectionDeepMapper = Mockito
+			.mock(IConceptCollectionDeepMapper.class);
+
+	@Mock
+	private IProjectShallowMapper mockedProjectShallowMapper = Mockito.mock(IProjectShallowMapper.class);
+
+	@Mock
+	private IListWsDAO mockedwsListManger = Mockito.mock(IListWsDAO.class);
+
+	@Mock
+	private IListWSManager mockedwsManager = Mockito.mock(IListWSManager.class);
+
+	@Mock
+	private IConceptFactory mockedConceptFactory = Mockito.mock(IConceptFactory.class);
+
+	@Mock
+	private IConceptCollectionDAO mockedccDao = Mockito.mock(IConceptCollectionDAO.class);
+
+	@InjectMocks
+	ConceptCollectionManager conceptCollectionManagerUnderTest;
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	private void mockRestTemplate(String methodName, Object returnType) {
+		if ("getForObject".equals(methodName)) {

```

Comment:
Hmm... I understand why you created this method, but you might just want to have 2 or 3 when-thenReturn statements in your setup method (since I believe you only have a few different cases). This way if someone accidentally breaks the call to the template by modifying the passed url, the tests will actually fail.